crates/storage/provider/src/bundle_state/state_reverts.rs

```diff
@@ -21,13 +21,27 @@ impl From<PlainStateReverts> for StateReverts {
 }
 
 impl StateReverts {
+    /// Write reverts to database, when [`StateReverts`] contains all values that will be inserted
+    /// for the given keys. See [`write_to_db_with_mode`](Self::write_to_db_with_mode).
+    pub fn write_to_db<TX: DbTxMut + DbTx>(
+        self,
+        tx: &TX,
+        first_block: BlockNumber,
+    ) -> Result<(), DatabaseError> {
+        self.write_to_db_with_mode(tx, first_block, true)
+    }
+
     /// Write reverts to database.
     ///
+    /// Passing `true` to to 'exhaustive_data_for_keys' parameter, will bring slight performance
+    /// gain, but will panic if values are not all that will be inserted for any given key.
+    ///
     /// Note:: Reverts will delete all wiped storage from plain state.
-    pub fn write_to_db<TX: DbTxMut + DbTx>(
+    pub fn write_to_db_with_mode<TX: DbTxMut + DbTx>(
         self,
         tx: &TX,
         first_block: BlockNumber,
+        exhaustive_data_for_keys: bool,
```

Comment:
@emhane `exhaustive_data_for_keys` parameter is unnecessary here. it can be replaced by looking at the last entry in the table and comparing it to the `first_block`